packages/browsers/test/src/firefox-data.spec.ts

```diff
@@ -69,4 +72,29 @@ describe('Firefox', () => {
       path.join('firefox', 'firefox.exe')
     );
   });
+
+  describe('profile', () => {
+    let tmpDir = '/tmp/puppeteer-browsers-test';
+
+    beforeEach(() => {
+      tmpDir = fs.mkdtempSync(
+        path.join(os.tmpdir(), 'puppeteer-browsers-test')
+      );
+    });
+
+    afterEach(() => {
+      fs.rmSync(tmpDir, {recursive: true});
+    });
+
+    it('should create a profile', async () => {
+      await createProfile(
+        {
+          test: 1,
+        },
+        tmpDir
+      );
+      const text = fs.readFileSync(path.join(tmpDir, 'user.js'), 'utf-8');
+      assert.ok(text.includes(`user_pref("test", 1);`));
```

Comment:
Nit: Let's assert a single default is set correctly, so we can be sure that we don't just discard them when you set a custom one.